spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/SparkBinPackPositionDeletesRewriter.java

```diff
@@ -107,12 +108,16 @@ protected void doRewrite(String groupId, List<PositionDeletesScanTask> group) {
 
     // keep only valid position deletes
     Dataset<Row> dataFiles = dataFiles(partitionType, partition);
-    Column joinCond = posDeletes.col("file_path").equalTo(dataFiles.col("file_path"));
+    Column joinCond =
+        posDeletes
+            .col(MetadataColumns.DELETE_FILE_PATH.name())
```

Comment:
mainly to make it easier to see that those columns refer to the ones from the `PositionDeletesTable`